test/jasmine/tests/indicator_test.js

```diff
@@ -0,0 +1,438 @@
+                type: 'indicator',
+                value: 1,
+                title: {text: 'Value'},
+                mode: 'number'
+            }])
+            .then(function() {
+                gd.style.display = 'block';
+
+                var t = d3.selectAll('text.title').node();
+                var titleBBox = t.getBoundingClientRect();
+
+                var numbers = d3.selectAll('text.numbers').node();
+                var numbersBBox = numbers.getBoundingClientRect();
+
+                expect(titleBBox.bottom).toBeCloseTo(numbersBBox.top - cn.titlePadding, 0);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('position it above angular axes', function(done) {
+            Plotly.newPlot(gd, [{
+                type: 'indicator',
+                value: 1,
+                title: {text: 'Value'},
+                mode: 'gauge',
+                gauge: {shape: 'angular'}
+            }])
+            .then(function() {
+                gd.style.display = 'block';
+                var t = d3.selectAll('text.title').node();
+                var titleBBox = t.getBoundingClientRect();
+
+                var ax = d3.selectAll('g.angularaxis').node();
+                var axBBox = ax.getBoundingClientRect();
+                expect(titleBBox.bottom).toBeCloseTo(axBBox.top - cn.titlePadding, 0);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('position it left of bullet', function(done) {
+            Plotly.newPlot(gd, [{
+                type: 'indicator',
+                value: 1,
+                title: {text: 'Value'},
+                mode: 'gauge',
+                gauge: {shape: 'bullet'}
+            }])
+            .then(function() {
+                gd.style.display = 'block';
+                var t = d3.selectAll('text.title').node();
+                var titleBBox = t.getBoundingClientRect();
+
+                var ax = d3.selectAll('g.bulletaxis').node();
+                var axBBox = ax.getBoundingClientRect();
+                expect(titleBBox.right < axBBox.left).toBe(true);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+    });
+
+    it('restyle between modes', function(done) {
+        function assertElementCnt(sel, cnt) {
+            var el = d3.selectAll(sel);
+            expect(el.size()).toBe(cnt, 'selection "' + sel + '" does not have size ' + cnt);
+        }
+        function assertGauge(shape, cnt) {
+            assertElementCnt('g.' + shape, cnt);
+            assertElementCnt('g.' + shape + 'axis', cnt);
+        }
+        function assert(flags) {
+            // flags is an array denoting whether the figure [hasNumber, hasDelta, hasAngular, hasBullet]
+            assertElementCnt('tspan.number', flags[0]);
+            assertElementCnt('tspan.delta', flags[1]);
+            assertGauge('angular', flags[2]);
+            assertGauge('bullet', flags[3]);
```

Comment:
Can we also assert that number, delta, angular and bullet pieces preserve their order after these `restyle` calls below.

You can try using 

https://github.com/plotly/plotly.js/blob/795220f815afcf79c73885715379b606afc1b4a1/test/jasmine/assets/custom_assertions.js#L309-L370

to help you out.

Reply:
It seems like the order is preserved as can be seen from the test introduced in commit 95de45724b62a7a1c3d22434af838bd8c690cbfe

Reply:
Same conclusion as in https://github.com/plotly/plotly.js/pull/3978#discussion_r298265531

Resolved.